hedera-node/src/main/java/com/hedera/services/state/expiry/EntityAutoRenewal.java

```diff
@@ -80,33 +83,40 @@ public void execute(Instant instantNow) {
 		if (networkCtxManager.currentTxnIsFirstInConsensusSecond()) {
 			curNetworkCtx.clearAutoRenewSummaryCounts();
 		}
-
-		renewalProcess.beginRenewalCycle(instantNow);
+		renewalProcess.beginRenewalCycle(nextAvailConsTime);
 
 		int i = 1;
 		int entitiesTouched = 0;
 		long scanNum = curNetworkCtx.lastScannedEntity();
-
-		log.debug("Auto-renew scan beginning at {}, wrapping at {}", scanNum, wrapNum);
+		boolean advanceScan = true;
+		EntityProcessResult result;
+		log.debug("Auto-renew scan beginning from last DONE @ {}, wrapping at {}", scanNum, wrapNum);
 		for (; i <= maxEntitiesToScan; i++) {
-			scanNum++;
-			if (scanNum >= wrapNum) {
-				scanNum = firstEntityToScan;
+			if (advanceScan) {
+				scanNum = next(scanNum, wrapNum);
 			}
-			if (renewalProcess.process(scanNum)) {
+			if ((result = renewalProcess.process(scanNum)) != NOTHING_TO_DO) {
 				entitiesTouched++;
+				advanceScan = (result == DONE);
+			} else {
+				advanceScan = true;
```

Comment:
This should be set to false ?

Reply:
If there's `NOTHING_TO_DO`, then we do want to advance our scan to the next entity.